test/e2e-samples/page-objects/router-sample-page.js

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint-disable */
+
+/**
+* main page object containing all methods, selectors and functionality
+* that is shared across all page objects
+*/
+class RouterSampleAppPage {
+
+  // Router App selectors
+  get header () { return $('main h1.page-header'); }
+  get homeNavLink () { return $('#home-nav-link'); }
+  get protectedNavLink () { return $('#protected-nav-link'); }
+  get loadingIcon () { return $('#loading-icon'); }
+  get footerBtn () { return $('footer > button[type="button"]'); }
+
+  async logout () {
+    await expect(this.footerBtn).toHaveText('Logout');
+    await this.footerBtn.click();
+    await browser.waitUntil(async () => await this.footerBtn.getText() === 'Login');
```

Comment:
nit: add timeoutMessage as a second arg in function for log debugging purpose, something like `wait for logout button`.